### ISSUE TITLE
Provide stack trace if a filter panics

### ISSUE THREAD

**Issue by maxim-tschumak** [zalando/skipper#1045]
**Is your feature request related to a problem? Please describe.**
In case a filter instance panics, we only know which filter failed (name), but we don't know where (file, line number, method, etc.)

**Describe the solution you would like**
A log output containing stack trace. Logging stack traces for all requests can be expensive. A good starting point would be to log the first panic stack trace only.

**Describe alternatives you've considered (optional)**
In order to not pollute the logs, we can implement a HTTP endpoint which can provide run-time information about Skipper (configured routes, errors, stats, etc.). This solution is more complex and requires further discussions.

**Additional context (optional)**
The `apiUsageMonitoringFilter` panics in one of our clusters. Without having this feature in place, we can only spot the problem by analyzing the code and/or trying to reproduce the error, which is less efficient.

**Would you like to work on it?**
Yes

**Comment by lmineiro** [comment]
I'm very confident that better unit testing of said filter can reveal the existing fragilities

**Comment by szuecs** [comment]
sgtm to log the panic.

In case of the alternative:
We have the support listener endpoint, defaults to :9911 which has:
- HEAD /routes  returns number of routes
- GET /routes returns the routes in eskip format
- GET /metrics returns metrics
- GET /debug/pprof to access via `go tool pprof` or browser 

@lmineiro I agree, but in this case https://codecov.io/gh/zalando/skipper/tree/master/filters shows that this filter has by far the most coverage from all the filters, which does not mean too much as we can see.

**Comment by maxim-tschumak** [comment]
@lmineiro I fully agree with you, we are already working on a PR where we want to check all the assumptions we made about the filter context and, of course, unit tests them.

@szuecs I also think that the proposed alternative is better. We've made good experience with providing such operational endpoint (`api-monitoring-controller` component). I think it doesn't contradict the PR I'm going to propose. I'm also happy to help with design/implementation of the operational endpoint :)
